crates/types/src/ycbcr422_image.rs

```diff
@@ -25,12 +25,34 @@ pub struct YCbCr422Image {
     buffer: Arc<Vec<YCbCr422>>,
 }
 
+impl From<RgbImage> for YCbCr422Image {
+    fn from(rgb_image: RgbImage) -> Self {
```

Comment:
What do you think about changing all uses of the `buffer_422_from_rgb_image()` and `rgb_image_from_buffer_422()` to uses of this trait? Maybe we could even inline the code of these functions into this trait to enforce usage of it? Currently there is some code duplication in the `impl YCbCr422Image` and this trait.

Reply:
Yeah I noted that too. I think i made some improvements in the edge detection branch, will check and patch this.

Reply:
Done: After looking at the buffer from/ to functions, I also noticed it is better to directly use the trait instead. Inlined those functions into the trait.